src/game/client/neo/ui/neo_hud_friendly_marker.cpp

```diff
@@ -150,133 +191,260 @@ void CNEOHud_FriendlyMarker::DrawPlayerForTeam(C_Team *team, const C_NEO_Player
+#endif //  GLOWS_ENABLE
+	const FriendlyMarkerInfo* settings = &m_szMarkerSettings[markerUsedIndex];
+
+	const Vector pos = player->GetAbsOrigin() + Vector(0, 0, player->GetPlayerMaxs().z + settings->iInitialOffset);
 	int x, y;
-	constexpr float HEIGHT_OFFSET = 56.0f;
-	auto pos = player->GetAbsOrigin() + Vector(0, 0, HEIGHT_OFFSET);
+	if (!GetVectorInScreenSpace(pos, x, y, nullptr)) {
+		return;
+	}
 
-	bool drawOutline = glow_outline_effect_enable.GetBool();
-	Vector offset = drawOutline ? Vector(0, 0, 7) : vec3_origin;
-	if (GetVectorInScreenSpace(pos, x, y, &offset))
-	{
-		const float fadeTextMultiplier = GetFadeValueTowardsScreenCentre(x, y);
-		if (fadeTextMultiplier > 0.001f)
-		{
-			static constexpr int MAX_MARKER_STRLEN = 48 + NEO_MAX_CLANTAG_LENGTH + 1;
-			const bool localPlayerAlive = const_cast<C_NEO_Player *>(localPlayer)->IsAlive();
-			const bool localPlayerSpec = (localPlayer->GetTeamNumber() < FIRST_GAME_TEAM);
-
-			surface()->DrawSetTextFont(m_hFont);
-			surface()->DrawSetTextColor(FadeColour(teamColor, fadeTextMultiplier));
-			int textYOffset = drawOutline ? vgui::surface()->GetFontTall(m_hFont) * -2 : 0;
-			char textASCII[MAX_MARKER_STRLEN];
-
-			auto DisplayText = [this, &textYOffset, x, y](const char *textASCII, bool drawOutline) {
-				wchar_t textUTF[MAX_MARKER_STRLEN];
-				g_pVGuiLocalize->ConvertANSIToUnicode(textASCII, textUTF, sizeof(textUTF));
-				int textWidth, textHeight;
-				surface()->GetTextSize(m_hFont, textUTF, textWidth, textHeight);
-				surface()->DrawSetTextPos(x - (textWidth / 2), y + (drawOutline ? 0 : m_iMarkerHeight) + textYOffset);
-				surface()->DrawPrintText(textUTF, V_wcslen(textUTF));
-				textYOffset += textHeight;
-			};
-
-			// Draw player's name and health
-			auto playerName = player->GetNeoPlayerName();
-
-			// Only show clan tag if spectator/no playing and this player has a clantag
-			const char *playerClantag = player->GetNeoClantag();
-			if (playerClantag && playerClantag[0] &&
-					(!cl_neo_clantag_friendly_marker_spec_only.GetBool() || localPlayerSpec))
-			{
-				V_sprintf_safe(textASCII, "[%s] %s", playerClantag, playerName);
-			}
-			else
-			{
-				V_strcpy_safe(textASCII, playerName);
-			}
-			DisplayText(textASCII, drawOutline);
+	char textASCII[MAX_MARKER_STRLEN];
+	int textSize = 0;
+
+	auto DisplayText = [this, &textSize, x](const char *textASCII, int y, int maxLength) {
+			wchar_t textUTF[MAX_MARKER_STRLEN];
+			COMPILE_TIME_ASSERT(sizeof(textUTF) == sizeof(wchar_t) * MAX_MARKER_STRLEN);
+			g_pVGuiLocalize->ConvertANSIToUnicode(textASCII, textUTF, min(sizeof(textUTF), sizeof(wchar_t) * maxLength));
+			int textWidth, textHeight;
+			vgui::surface()->GetTextSize(m_hFont, textUTF, textWidth, textHeight);
+			vgui::surface()->DrawSetTextPos(x - (textWidth / 2), y - textHeight);
+			vgui::surface()->DrawPrintText(textUTF, min(V_wcslen(textUTF), maxLength));
```

Comment:
Nit: we can get the text length from the `g_pVGuiLocalize->ConvertANSIToUnicode` return value (minus 1 to omit the null terminator). So we can skip the `min(V_wcslen(...` calculation. Assuming the API comment for the function is accurate.